api/src/opentrons/protocol_api_experimental/module_contexts/heater_shaker_module_context.py

```diff
@@ -0,0 +1,83 @@
+"""Protocol API interface for Heater-Shaker Modules."""
+
+from ..errors import (
+    InvalidTargetTemperatureError,
+    InvalidTargetSpeedError,
+)
+
+TEMPERATURE_LOWER_LIMIT = 37
+TEMPERATURE_UPPER_LIMIT = 95
+SPEED_LOWER_LIMIT = 200
+SPEED_UPPER_LIMIT = 3000
+
+
+class HeaterShakerModuleContext:
+    # TODO(spp, 2022-02-15): Revise this when writing other module contexts' docstrings.
+    """Object representing a connected Heater-Shaker Module.
+
+    It should not be instantiated directly; instead, it should be created through
+    :py:meth:`.ProtocolContext.load_module`.
+    """
+
+    def __init__(self, module_id: str) -> None:
+        self._module_id = module_id
+
+    def start_set_temperature(self, temperature: float) -> None:
+        """Set the target temperature of heater-shaker and return immediately.
+
+        This method takes a target temperature between 37C & 95C and tells
+        the heater-shaker to start heating to the target temperature without waiting
+        for the heater-shaker to reach the target.
+        Use :py:meth:`await_temperature` to wait for the temperature set here.
+
+        Raises:
+            InvalidTargetTemperatureError: target temperature out of limits.
+        """
+        if not TEMPERATURE_LOWER_LIMIT <= temperature <= TEMPERATURE_UPPER_LIMIT:
+            raise InvalidTargetTemperatureError(
+                f"Temperature should be in range {TEMPERATURE_LOWER_LIMIT} to "
+                f"{TEMPERATURE_UPPER_LIMIT} degree celsius."
+            )
+        raise NotImplementedError()
+
+    def await_temperature(self) -> None:
+        """Wait for the heater-shaker to reach its target temperature.
+
+        Use :py:meth:`start_set_temperature` to set the target temperature first.
+
+        Raises:
+            NoTargetTemperatureError: If heater-shaker has no target temperature.
+        """
+        raise NotImplementedError()
+
+    def stop_heating(self) -> None:
+        """Stop heating."""
+        raise NotImplementedError()
+
+    def set_shake(self, speed: int) -> None:
+        """Set shake speed in RPM and start shaking.
+
+        Latches the labware, starts shaking the plate and returns once the target
+        shake speed has reached.
+        Acceptable shake speed: 200 - 3000 RPM
+
+        Raises:
+            InvalidTargetSpeedError: if target speed out of limits.
+        """
```

Comment:
If we wanted, we could do this:

```suggestion
        f"""Set shake speed in RPM and start shaking.

        Latches the labware, starts shaking the plate and returns once the target
        shake speed has reached.
        Acceptable shake speed: {SPEED_LOWER_LIMIT} - {SPEED_UPPER_LIMIT} RPM

        Raises:
            InvalidTargetSpeedError: if target speed out of limits.
        """
```

Same for the temperature limits.

Reply:
Turns out, we [can't really do that](https://stackoverflow.com/questions/36705130/python-docstrings-templated). The docstrings are not _executed_ when a function is created so the variables aren't resolved. There's a way to add formatted strings to docstrings but that requires us to assign the string to a function's `__doc__` attribute which is kinda ugly.